spec/pointer_spec.rb

```diff
@@ -0,0 +1,12 @@
+RSpec.describe Kalculator do
+  describe "Pointers" do
+    it "points to another variable in the data sources" do
+      data = Kalculator::DataSources.new({"a"=>Kalculator::Pointer.new("name"), "name"=>"bob"})
```

Comment:
Just a thought.... would it be possible to set this up as something like:

```ruby
jan = {"name" => "Jan", "age" => 22}
susan = {"name" => "Susan", "age" => 50, "daughter" => jan}
bob = {"name" => "Bob", "age" => 20, "mother" => Pointer.new(susan)}
data_source = {"bob" => bob, "susan" => susan}
Kalculator.evaluate("bob.mother.name", data_source) # => "Susan"
Kalculator.evaluate("bob.mother.daughter.name", data_source) # => "Jan"
```

Basically allow you to point to an arbitrary piece of data that doesn't necessarily have its own top-level name? I think the tricky part of that is the type-checking? Not sure what the impact would be, but I think we have this use-case in the Spiff product

Reply:
I don't quite get what you are asking, in the example you pointed out, if you replace Pointer.new(susan) with Pointer.new("susan") it will still run correctly. However it will not run if "susan" => susan is not in the data sources. 

to get the same effect as in your example all you have to do is
`jan = {"name" => "Jan", "age" => 22}
susan = {"name" => "Susan", "age" => 50, "daughter" => jan}
bob = {"name" => "Bob", "age" => 20, "mother" => Pointer.new("susan")}
data_source = {"bob" => bob, "susan" => susan}
Kalculator.evaluate("bob.mother.name", data_source) # => "Susan"
Kalculator.evaluate("bob.mother.daughter.name", data_source) # => "Jan"`


currently pointers arent able to store data that is referenced in the data sources file

Reply:
*isn't referenced in the data sources

(it's still pretty early for me XD)

Reply:
I'm thinking of the use-case where a piece of data doesn't have a global name (ie a key at the root of the data source). Imagine a case where we are loading in a deal with fields like `{"amount" => 100.0, "name" => "Xerox", "account" => ...}`, the account field would point to an account record. It's just one of many different accounts for this company and doesn't have a specific name in the data source. Could I still make a pointer to that account record?

Reply:
Oh! you wouldn't be able to do that currently, lemme add that.

Reply:
Ok that has been added as an AnonymousPointer, it works like 
`{"amount" => 100.0, "name" => "Xerox", "account" => Kalculator::AnonymousPointer.new("bob")}` 
where "bob" could be any data at all

works the same with type inference

Reply:
you would get the same effect as having this pointer if you just put the data directly into the data sources like so

`account = some-anonymous-data`
` {"amount" => 100.0, "name" => "Xerox", "account" => account`

Reply:
oop forgot a curly brace. `}`. There you go